src/seedu/addressbook/data/person/Address.java

```diff
@@ -21,38 +26,63 @@
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+	this.isPrivate = isPrivate;
+        
+	String[] components = address.split(",");
+	int lenComponents = components.length;
+	for (int i = lenComponents; i < NUM_COMPONENTS; i++) {
+	    components[i] = null;
+	}
+
+	this.block = new Block(components[0]);
+	this.street = new Street(components[1]);
+	this.unit = new Unit(components[2]);
+	this.postalCode = new PostalCode(components[3]);
+	
+	if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
```

Comment:
Think about whether this is the best place to do validation. Is your added code able to handle the exceptions?